internal/elasticsearch/index/indices/models.go

```diff
@@ -284,6 +284,20 @@ func setSettingsFromAPI(ctx context.Context, model *indexTfModel, apiModel model
 			}
 			tfValue = basetypes.NewStringValue(settingStr)
 		case types.Bool:
+			if settingStr, ok := settingsValue.(string); ok {
+				settingBool, err := strconv.ParseBool(settingStr)
+				if err != nil {
+					return diag.Diagnostics{
+						diag.NewErrorDiagnostic(
+							"failed to convert setting to bool",
+							fmt.Sprintf("expected setting to be a bool but it was a string. Attempted to parse it but got %s", err.Error()),
+						),
+					}
+				}
+
+				settingsValue = bool(settingBool)
```

Comment:
```suggestion
				settingsValue = settingBool
```

Reply:
Thanks a lot @tobio for your fast review.

My bad for the changelog, I forgot about...
I just pushed a new commit with the changelog updated and your code suggestion applied. Thanks.